### ISSUE TITLE
How write Keras custom metric not using y_true and y_pred

### ISSUE THREAD

**Issue by shichengcn** [keras-team/keras#8722]
I use keras to implement the model in the paper "Comparative Deep Learning of Hybrid Representations".

which uses triple input for training(high quality pictures, low quality pictures, user vectors) .

double input for testing (picture sequence, user vector)

By training I want to make the distance between user vectors and good pictures is much smaller than the distance between user vectors and bad pictures. The network of good and bad pictures are exactly the same.

the distance between a user vectors and  a group of picture vectors  can be calculated for recommendation.So the model just need double input for testing.

I have to write the custom Keras metric function.The problem is that  y_true and y_pred can not be used in this case and part of the model is to be used to calculate the distance between images and users.（ there is no good or bed pictures in the process of verification）

## this is basic concepts of my model：
```
def CML_Model([good_image,bed_image], User_vec):
    good_image_vec = vision_model(good_image)
    bed_image_vec = vision_model(bed_image)
               # the network are exactly the same.
    User_vec = user_model(test_User)   
    Calculating_distance_a(User_vec ,good_image_vec )
    Calculating_distance_b(bed_image_vec  ,good_image_vec )
     loss = f(Calculating_distance_b - Calculating_distance_a)
     #do a mapping f ，then i got loss，the greater the difference between the distance, the closer the loss to 1
```
By training it can make the distance between user vectors and good pictures is much smaller than the distance between user vectors

## this is basic concepts of my custom metric function：

There are some good pictures and bad pictures in the input pictures. I want to calculate  and sort the distance between all pictures and user vectors, and calculate whether the good pictures are closer to the users, and get the accuracy rate.
```
def CML_top_k_accracy(y_true, y_pred):                  #i want input （group of pictures，user vector）
    image_vec = vision_model(group of picture)         #part of the current model
    User_vec = user_model(user vector)                 #part of the current model
    ...Calculating distance and sorting
    ...Calculation accuracy
```
The models of training and verification are slightly different and the input and output are different
How can I use a part of the current model weights to calculate my_acc in custom metric function？

code：https://github.com/shichengcn/ImageAestheticAssessment/blob/master/CDL_dual_net.py
